### PR TITLE
W-13676156: changes to adapt project to IntelliJ

### PR DESCRIPTION
**Pull request by mbuchwald**
Make Intellij Great Again. 

The main goal was to make that java11 directories to not depend on the java 8 directories (full explanation [here](https://docs.google.com/document/d/1T2p_n_oNh-rio_Ru9f4LDO5ukX5hed6xJ5GR-IZW5Qs/edit#heading=h.pvww9i63jif)) + script to change imls.

### CODE REVIEW

File: modules/service-artifact-impl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.mule.runtime</groupId>
+        <artifactId>mule-modules</artifactId>
+        <version>4.5.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>mule-module-service-artifact-impl</artifactId>
+    <name>Mule Services Artifact Implementation</name>
+    <description>BLA BLA BLA</description>
```
Comment (marianogonzalez) [inline modules/service-artifact-impl/pom.xml:12]: I thnk you have a typo on the second BLA

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/service-artifact-impl/pom.xml (hunk 1 above)
Comment (mbuchwald) [reply modules/service-artifact-impl/pom.xml]: No, but I do in the third one, heh

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Comment by mbuchwald** [comment]
--validate

**Review by rbourbonnavarro — APPROVED** [review]

File: config-intelliJ.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+if [ "$1" = "java8" ]; then
```
Comment (marianogonzalez) [inline config-intelliJ.sh:1]: @elrodro83 , Martin's opinion is that this script should be here. I think it should go in mule-uber/bin as all the other scripts. What say you?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/service-artifact-impl/src/main/java11/org/mule/runtime/module/service/api/artifact/impl/ServiceClassLoaderFactoryProvider.java [hunk 3]
```diff
@@ -1,10 +1,12 @@
 /*
  * Copyright 2023 Salesforce, Inc. All rights reserved.
  */
-package org.mule.runtime.module.service.api.artifact;
+package org.mule.runtime.module.service.api.artifact.impl;
```
Comment (marianogonzalez) [inline modules/service-artifact-impl/src/main/java11/org/mule/runtime/module/service/api/artifact/impl/ServiceClassLoaderFactoryProvider.java:4]: isn't this a breaking change?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: config-intelliJ.sh (hunk 2 above)
Comment (elrodro83) [reply config-intelliJ.sh:1]: vote for `mule-uber/bin`, as all other tools used to set up the workspace

File: modules/service-artifact-impl/src/main/java11/org/mule/runtime/module/service/api/artifact/impl/ServiceClassLoaderFactoryProvider.java (hunk 3 above)
Comment (elrodro83) [reply modules/service-artifact-impl/src/main/java11/org/mule/runtime/module/service/api/artifact/impl/ServiceClassLoaderFactoryProvider.java:4]: why is this needed?

**Pull request closed without merging**